### PR TITLE
docs(tutorial): Adds more TS hints in the toturial

### PR DESCRIPTION
**Pull request by dac09**
People following the tutorial seem to be getting caught out quite a lot by the TS errors. 

This adds some extra tips to help deal with the situation. 

@jtoar feel free to edit the copy as you see fit!

### CODE REVIEW

File: docs/docs/tutorial/chapter7/rbac.md [hunk 1]
```diff
@@ -96,6 +96,74 @@ export const getCurrentUser = async (session) => {
...
+```diff title="api/src/lib/auth.ts"
+export const hasRole = (roles: AllowedRoles): boolean => {
+  if (!isAuthenticated()) {
+    return false
+  }
+
+  const currentUserRoles = context.currentUser?.roles
+
+  if (typeof roles === 'string') {
+    if (typeof currentUserRoles === 'string') {
```
Comment (jtoar) [inline docs/docs/tutorial/chapter7/rbac.md:129]: Do we need this if statement if we know `currentUserRoles` is a string?

File: docs/docs/tutorial/chapter7/rbac.md [hunk 2]
```diff
@@ -96,6 +96,74 @@ export const getCurrentUser = async (session) => {
...
+      return currentUserRoles?.some((allowedRole) =>
+        roles.includes(allowedRole)
+      )
+    } else if (typeof context?.currentUser?.roles === 'string') {
+      // roles to check is an array, currentUser.roles is a string
+      return roles.some(
+        (allowedRole) => context.currentUser?.roles === allowedRole
+      )
+    }
+  }
```
Comment (jtoar) [inline docs/docs/tutorial/chapter7/rbac.md:150]: Should this block be changed as well?

**Review by jtoar — COMMENTED** [review]
@dac09 made a few edits and have a few questions about making more changes to the RBAC section, but otherwise looks good

File: docs/docs/tutorial/chapter7/rbac.md (hunk 1 above)
Comment (dac09) [reply docs/docs/tutorial/chapter7/rbac.md:129]: Yeah - so we definitely don't need this code - and my first plan was to just copy paste these changes https://redwoodjs.com/docs/canary/typescript/strict-mode#b-roles-on-current-user-is-a-string - but it really does make it seem scary. 

So to keep it simple, I'm only suggesting the bare minimum to make the red squigglies go away - which seems to be the main problem that some people report. Often people aren't reading the code - just get scared when they see the red.

File: docs/docs/tutorial/chapter7/rbac.md (hunk 2 above)
Comment (dac09) [reply docs/docs/tutorial/chapter7/rbac.md:150]: same as above.

File: docs/docs/tutorial/chapter2/cells.md [hunk 3]
```diff
@@ -318,6 +318,24 @@ export const Success = ({ posts }: CellSuccessProps<ArticlesQuery>) => {
 </TabItem>
 </Tabs>
 
+<ShowForTs>
+
+:::tip Using generated types
+
+At this point, you might see an error in your Cell while trying to import from `types/graphql`: "The type ArticlesQuery does not exist"
+
+When you have `yarn rw dev` running, the CLI watches files for changes and triggers type generation automatically, but you can trigger it manually too by running:
```
Comment (jtoar) [inline docs/docs/tutorial/chapter2/cells.md:327]:
```suggestion
When you have the dev server (via `yarn rw dev`) running, the CLI watches files for changes and triggers type generation automatically, but you can trigger it manually too by running:
```

File: docs/versioned_docs/version-2.2/tutorial/chapter2/cells.md [hunk 4]
```diff
@@ -318,6 +318,24 @@ export const Success = ({ posts }: CellSuccessProps<ArticlesQuery>) => {
 </TabItem>
 </Tabs>
 
+<ShowForTs>
+
+:::tip Using generated types
+
+At this point, you might see an error in your Cell while trying to import from `types/graphql`: "The type ArticlesQuery does not exist"
+
+When you have `yarn rw dev` running, the CLI watches files for changes and triggers type generation automatically, but you can trigger it manually too by running:
```
Comment (jtoar) [inline docs/versioned_docs/version-2.2/tutorial/chapter2/cells.md:327]:
```suggestion
When you have the dev server (via `yarn rw dev`) running, the CLI watches files for changes and triggers type generation automatically, but you can trigger it manually too by running:
```

File: docs/docs/tutorial/chapter7/rbac.md [hunk 5]
```diff
@@ -96,6 +96,66 @@ export const getCurrentUser = async (session) => {
 </TabItem>
 </Tabs>
 
+<ShowForTs>
+
+### Fixing the hasRole function
+
+At this point, you might notice an error in your `api/src/lib/auth.ts` file, in the `hasRole` function. TypeScript is trying to help you here, by highlighting that roles can never be an an array of strings:
```
Comment (jtoar) [inline docs/docs/tutorial/chapter7/rbac.md:103]:
```suggestion
At this point, you might notice an error in your `api/src/lib/auth.ts` file, in the `hasRole` function. TypeScript is trying to help you here, by highlighting that roles can never be an array of strings:
```

File: docs/versioned_docs/version-2.2/tutorial/chapter7/rbac.md [hunk 6]
```diff
@@ -96,6 +96,66 @@ export const getCurrentUser = async (session) => {
 </TabItem>
 </Tabs>
 
+<ShowForTs>
+
+### Fixing the hasRole function
+
+At this point, you might notice an error in your `api/src/lib/auth.ts` file, in the `hasRole` function. TypeScript is trying to help you here, by highlighting that roles can never be an an array of strings:
```
Comment (jtoar) [inline docs/versioned_docs/version-2.2/tutorial/chapter7/rbac.md:103]:
```suggestion
At this point, you might notice an error in your `api/src/lib/auth.ts` file, in the `hasRole` function. TypeScript is trying to help you here, by highlighting that roles can never be an array of strings:
```

**Review by jtoar — APPROVED** [review]
Few more edits, and approved

**Pull request merged**